### ISSUE TITLE
Minestom

### ISSUE THREAD

**Issue by Dqgs** [LuckPerms/LuckPerms#3077]
### Description

Able to use Luckperms with Minestom

### Proposed behaviour

This will have to ability to use Minestom.

**Comment by lucko** [comment]
I'll accept a good PR implementing this.

**Comment by Dqgs** [comment]
Someone attempted to make a fork already you could look at what they did https://github.com/Project-Cepi/LuckPerms

Disclaimer i just found this and i did 0 work into this. This may not be fully done and credit should go where credit is due.

**Comment by darkinfire** [comment]
Hi, will it ever be added?

**Comment by xSehrMotiviert** [comment]
Would be nice. I'm using the fork, but it's not really working for me. Maybe because I use the 1.19 version of Minestom. 😅

**Comment by zax71** [comment]
#3519 could be promising

**Comment by General-Zimmer** [comment]
> #3519 could be promising
^ this was moved to https://github.com/LuckPerms/LuckPerms/pull/3521

**Comment by Bloeckchengrafik** [comment]
Hey everyone,
this might be a topic worth revisiting since minestom [moved away from jitpack to maven central](https://github.com/Minestom/Minestom/pull/1937). Also, there is [this](https://github.com/hollow-cube/minestom-ce/pull/105) PR by @Codestech1 that might come in handy while working on an official integration. Although the last linked pr is currently only on the (now deprecated) ce-Repository, but a rebase _should_ be possible fairly easily.

**Comment by ajh123** [comment]
Good news everyone! The main [Minestom repo](https://github.com/Minestom/Minestom) has had the changes from [Minestom CE](https://github.com/hollow-cube/minestom-ce) merged. That also means the normal Minestom is now available on Maven central.

**Comment by Laarryy** [comment]
Minestom removed permissions in https://github.com/Minestom/Minestom/pull/2445 so this is likely to go nowhere.

**Comment by LooFifteen** [comment]
The purpose of the removal is quite the opposite. The intention was to allow more freedom and make it easier to integrate things like LuckPerms into Minestom, without the restrictions of the previous permission system. This way, servers should directly communicate with their permission handler instead of going through a limited middleman.

**Comment by ajh123** [comment]
> The purpose of the removal is quite the opposite. The intention was to allow more freedom and make it easier to integrate things like LuckPerms into Minestom

Exactly, the old permission system couldn't represent a TriState now we have complete control by manually checking for valid permissions in the command handler (which means that commands must opt into permission handling).

Even Luck Perms was cited as a reason to remove the built in system:

> The built-in permissions system is too raw, it doesn't contain tristate and doesn't allow you to conveniently connect an external provider like LuckPerms
> from: https://github.com/Minestom/Minestom/pull/2302

**Comment by Laarryy** [comment]
That's great! I'll reopen this in the hopes that someone sees and PRs a Minestom implementation for LP. As Luck said, it would be welcome, and I think that given the removal of the middleman (as so kindly pointed out) will probably make supporting LP easier! :D
